overlord/devicestate/devicemgr.go

```diff
@@ -931,6 +948,22 @@ func (m *DeviceManager) ensureInstalled() error {
 	}
 	prev = setupRunSystem
 
+	// add the install-device hook before ensure-next-boot-to-run-mode if it
+	// exists in the snap
+	if hasInstallDeviceHook {
+		summary := i18n.G("Run install-device hook")
+		hooksup := &hookstate.HookSetup{
+			// TODO: what's a reasonable timeout for the install-device hook?
+			Snap: model.Gadget(),
+			Hook: "install-device",
+		}
+		installDevice := hookstate.HookTask(m.state, summary, hooksup, nil)
+		addTask(installDevice)
+		// hooks are under a different manager, make sure we consider
+		// it immediately
```

Comment:
this is not true anymore, there is only one taskrunner configured after all the other managers, so likely don't need the EnsureBefore anymore, probably same for a couple of others in this file

Reply:
thanks for pointing this out, I will clean this up in the other instances too